ui-v2/app/services/settings.js

```diff
@@ -7,19 +7,27 @@ export default Service.extend({
   storage: window.localStorage,
   findHeaders: function() {
     // TODO: if possible this should be a promise
+    const token = get(this, 'storage').getItem('token');
+    // TODO: The old UI always sent ?token=
+    // replicate the old functionality here
```

Comment:
FWIW `X-Consul-Token` is the right way to do it in this scenario, instead of passing `?token`.

Reply:
Hey, sorry yeah should have explained that comment a bit better, right now I'm replicating the fact that the old UI used to send the `?token` with *every single request*. So I'm doing the same currently but with `X-Consul-Token` instead of `?token` (so replicating the fact that we send it with every single request). I was going to look at whether that is necessary to send it with every single request further down the line, the more I think about it though we probably do.